tests/compiletest.rs

```diff
@@ -68,18 +73,20 @@ fn run_tests(mode: Mode, path: &str, target: Option<String>) -> Result<()> {
         path_filter: path_filter.collect(),
         program: miri_path(),
         output_conflict_handling,
-        dependencies_crate_manifest_path: use_std
+        dependencies_crate_manifest_path: (with_dependencies && use_std)
             .then(|| Path::new("test_dependencies").join("Cargo.toml")),
-        dependency_builder: Some(DependencyBuilder {
-            program: std::env::var_os("CARGO").unwrap().into(),
-            args: vec![
-                "run".into(),
-                "--manifest-path".into(),
-                "cargo-miri/Cargo.toml".into(),
-                "--".into(),
-                "miri".into(),
-            ],
-            envs: vec![],
+        dependency_builder: with_dependencies.then(|| {
```

Comment:
Why does this use a different boolean than `dependencies_crate_manifest_path`? Why are these even two different fields, does it ever make sense to have one be `Some` and the other `None`?

Reply:
for clippy this would be `None` as it would just use default `cargo`. Of course we could say that if you add a manifest path, you also need a command for it, but I would expect almost no one but miri needing this.

Reply:
That doesn't answer the first question though -- why is it `(with_dependencies && use_std)` there and just `with_dependencies` here? We can just keep this as `Some(...)`, right, since it will be ignored anyway when `dependencies_crate_manifest_path` is `None`? We should either do that or use the same condition for both.

Reply:
I switched back to the previous logic. I just thought it may be useful to be consistent I think, but you're right, this can be set and we can just ignore it